src/Symfony/Component/Validator/Mapping/Loader/AbstractLoader.php

```diff
@@ -78,6 +79,10 @@ protected function newConstraint(string $name, mixed $options = null): Constrain
             $className = self::DEFAULT_NAMESPACE.$name;
         }
 
+        if ([] !== (new \ReflectionClass($className))->getAttributes(NamedArgumentConstraintConstructor::class) && \is_array($options)) {
```

Comment:
it'd remove the is_array check because the attribute has to be enough to tell how to create the constraint

also, how perf-critical is this piece of code? do we need a cache?, eg
`if ($this->namedArgumentsCache[$className] ??= (bool) (new \ReflectionFunction($className, '__construct'))->getAttributes(HasNamedArguments::class)) {`

Reply:
updated